ui/src/containers/App.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect } from 'react';
+import { BrowserRouter as Router, Switch, Redirect, Route } from 'react-router-dom';
+import LandingPage from '../components/LandingPage';
+import PropTypes from 'prop-types';
+import AppPageContainer from './appView/AppPageContainer';
+import { authenticate } from '../actions/actions-ui';
+import { connect } from 'react-redux';
+import config from '../config/config';
+
+/**
+ * Main entry point for the entire UI view.
+ *
+ * @param {func} authenticate - Redux action to check if the user is logged in.
+ * @param {bool} isLoggedIn - Whether the user is logged in or not.
+ * @param {bool} isLoading - Whether the component has finished loading.
+ */
+export const App = ({ authenticate, isLoggedIn, isLoading }) => {
+  useEffect(() => {
+    authenticate();
+  }, [authenticate]);
+
+  /**
+   * Check to see if the user is logged in.
+   * If not, redirect them to the login screen.
+   */
+  const checkIfAuthenticated = () => {
+    if (config.mode === 'production' && !isLoading && !isLoggedIn) {
+      window.location.replace(config.auth.signInUrl);
```

Comment:
I'm wondering if we should use `window.location.assign` here instead? Effect is the same, but using `replace` we lose the ability to use the back button on the browser

Reply:
I will try it out. I tried [`react-router/Redirect`](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Redirect.md) before and that did not work.

Reply:
I have tried this out and can confirm it will not work as we want. The problem is that `window.location.assign` will allow a user to use the back button to get back to the home page, even if they are not logged in.

Reply:
And if they are not logged in it should not matter because they should not be able to use the back button to navigate to it.

Reply:
Yes agreed, we should use the solution which blocks the user from being able to use the back button which would show them the UI without data.

Reply:
Re-opening this for further clarification - if our approach here to disable the back button so the user cannot navigate back to that page, what stops the user from manually entering the URL & navigating back to that view and being able to see the UI?

- it seems disabling the back button is security by obscurity which is not recommended